src/macros.jl

```diff
@@ -1322,6 +1330,9 @@ macro objective(model, args...)
     newaff, parsecode = _MA.rewrite(x)
     code = quote
         $parsecode
+        # Don't leak a `_MA.Zero` if the objective expression is an empty
+        # summation, or other structure that returns `_MA.Zero()`.
+        $newaff = _replace_zero($newaff)
```

Comment:
Pourquoi ajouter de la complexité dans le macro code au lieu de garder la méthode pour `set_objective_function` ?

Reply:
Because it returns the `newaff` below. So `obj = @objective(model, Min, sum(i for i in 1:0))` leaks.